### ISSUE TITLE
Enhancement : data depth compatibility

### ISSUE THREAD

**Issue by rsepierre** [magynhard/curly-bracket-parser#3]
I think it would be nice to add the possibility to allow depth in the provided data object.

For exemple  `{{resume.title}}`, `{{figure.slider.description}}` could be replace with the content of `passedData['resume']['title']` and `passedData['figure']['slider']['description']` respectively.

For myself, I'm trying to parse a document with an object containing data at varying depth of an object tree.

Currently I have to either flatten my data object, or split the parsing of my page into 6 different steps. And I actually have to add another step on top of it, reading the whole document, because I need one data element multiple times accross the whole document. I can't do at the moment simply due to how the data is arranged.

This is the current exemple 

(excerpt of exemple data)
![image](https://user-images.githubusercontent.com/6746875/194581875-66895548-b8db-44d6-95ba-060d2833e25f.png)
(excerpt of exemple javascript)
![image](https://user-images.githubusercontent.com/6746875/194582860-cc755279-e8a7-4f66-a5d9-2863b231a147.png)
(excerpt of exemple html)
![image](https://user-images.githubusercontent.com/6746875/194583223-7fe9b676-be49-4254-86a5-57942b2daa31.png)

With this enhancement, I could parse the whole document at once, in one command, without modifying how my data is stored.

I think it would be very useful in and of itself but would also probably open possibilities for loop filters, and other neat things down the road.

**Comment by magynhard** [comment]
Hello @rsepierre, thank you for your request!

I like the feature you requested and will check to implement it soon!

**Comment by magynhard** [comment]
Hello @rsepierre,

i just released 1.3.1 with support for tree variables. Please try it and give me a short feedback to ensure if it suits all your needs!

**Comment by rsepierre** [comment]
Ah nice, super cool !
I've implemented my own parsing for the project but think I will add the library back in now !

**Comment by rsepierre** [comment]
works like a charm. Love it

**Comment by magynhard** [comment]
Thank you! :+1:

**Comment by rsepierre** [comment]
Hello @magynhard .

I think I found a small issue with this update.
When tree variables are not found it throws an error which states that "TreeVariableNotFoundError" doesn't exist.
I can confirm that I can't find the `TreeVariableNotFoundError` Class

![image](https://user-images.githubusercontent.com/6746875/208418996-a37f0461-5027-4fe1-89d9-ed35108a68ab.png)

Adding the error like the others fixes the issue.
```
class TreeVariableNotFoundError extends Error {
	constructor(message) {
		super(message);
		this.name = "TreeVariableNotFoundError";
	}
}
```

**Comment by rsepierre** [comment]
I would do a pull request, but I don't use same minifier as you and I use tabulations so my commits are a bit of a mess 😄 
https://github.com/rsepierre/curly-bracket-parser/tree/treevariable_error_fix

**Comment by magynhard** [comment]
@rsepierre thank you for your feedback. Should be fixed now in release 1.3.3. Can you confirm that your problem is solved?
